// vue-tg/dist/index.js
import { defineComponent as r, onMounted as u, watch as i, nextTick as C, onBeforeUnmount as g, computed as v, createBlock as d, openBlock as p, resolveDynamicComponent as m, withCtx as A, ref as x } from "vue";
import { n as _, a as w, b as P, m as h, t as k, v as T, l as W, o as E, p as q, r as M, w as L, x as I, y as D, z as j, A as F, B as $, c as R, C as O, d as U, i as Q, D as z, E as H, F as N, G, H as J, I as K, J as X, K as Y, s as Z, L as ee } from "./useSecureStorage-CSgDjq9B.mjs";
import { M as Re, Y as Oe, Z as Ue, W as Qe, X as ze, N as He, Q as Ne, O as Ge, a0 as Je, a1 as Ke, _ as Xe, $ as Ye, ac as Ze, ab as et, aa as tt, ad as st, S as ot, T as nt, a4 as at, a5 as it, a2 as rt, a3 as ct, U as ut, V as lt, a6 as dt, a7 as pt, P as mt, R as gt, a9 as ft, a8 as ht } from "./useSecureStorage-CSgDjq9B.mjs";
import { useAccelerometer as Bt, useBackButton as yt, useBiometricManager as bt, useClipboard as Ct, useCloudStorage as vt, useDeviceOrientation as At, useDeviceStorage as _t, useEmojiStatus as kt, useGyroscope as xt, useHapticFeedback as wt, useHomeScreen as Pt, useLocationManager as Tt, useMainButton as Wt, useMiniApp as Et, usePopup as Mt, useQrScanner as Lt, useSecondaryButton as Vt, useSecureStorage as qt, useSettingsButton as It, useTheme as Dt, useViewport as jt } from "./6.0.js";
const te = /* @__PURE__ */ r({
  __name: "Alert",
  props: {
    message: {
      type: String,
      required: !0
    }
  },
  emits: ["close"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = _("6.0");
    return s.isVersionAtLeast("6.2") && u(() => {
      s.showAlert(t.message, () => a("close"));
    }), (o, c) => null;
  }
}), se = /* @__PURE__ */ r({
  __name: "BackButton",
  props: {
    visible: { type: Boolean, default: !0 }
  },
  emits: ["click"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = w("6.0");
    return s.isVersionAtLeast("6.1") && (i(
      () => t.visible,
      (o) => {
        o ? s.show() : s.hide();
      }
    ), s.onClick(() => a("click")), u(() => {
      t.visible && C(() => {
        s.show();
      });
    }), g(() => s.hide())), (o, c) => null;
  }
}), oe = /* @__PURE__ */ r({
  __name: "BiometricManager",
  emits: ["init"],
  setup(e, { emit: n }) {
    const t = n, a = P("6.0");
    return a.isVersionAtLeast("7.2") && u(() => a.init(() => t("init"))), (s, o) => null;
  }
}), ne = /* @__PURE__ */ r({
  __name: "ClosingConfirmation",
  setup(e) {
    const n = h("6.0");
    return n.isVersionAtLeast("6.2") && (u(() => {
      n.isClosingConfirmationEnabled.value = !0;
    }), g(() => {
      n.isClosingConfirmationEnabled.value = !1;
    })), (t, a) => null;
  }
}), ae = /* @__PURE__ */ r({
  __name: "Confirm",
  props: {
    message: { type: String, required: !0 }
  },
  emits: ["close"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = _("6.0");
    return s.isVersionAtLeast("6.2") && u(() => s.showConfirm(t.message, (o) => a("close", o))), (o, c) => null;
  }
}), ie = /* @__PURE__ */ r({
  __name: "ExpandedViewport",
  props: {
    force: { type: Boolean, default: !1 }
  },
  setup(e) {
    const n = e, t = k("6.0");
    return t.isVersionAtLeast("7.7") ? (u(() => {
      t.isVerticalSwipesEnabled.value = !1;
    }), g(() => {
      t.isVerticalSwipesEnabled.value = !0;
    })) : (i(
      () => n.force,
      (a) => {
        a && !t.isExpanded.value && t.expand();
      },
      { immediate: !0 }
    ), T(({ isStateStable: a }) => {
      n.force && a && !t.isExpanded.value && t.expand();
    })), u(() => t.expand()), (a, s) => null;
  }
}), re = /* @__PURE__ */ r({
  __name: "FullscreenViewport",
  setup(e) {
    const n = k("6.0");
    return n.isVersionAtLeast("8.0") && (u(() => {
      n.isFullscreen.value = !0;
    }), g(() => {
      n.isFullscreen.value = !1;
    })), (t, a) => null;
  }
}), ce = /* @__PURE__ */ r({
  __name: "MainButton",
  props: {
    text: { type: String },
    color: { type: String },
    textColor: { type: String },
    visible: { type: Boolean, default: !0 },
    disabled: { type: Boolean, default: !1 },
    progress: { type: Boolean, default: !1 },
    hasShineEffect: { type: Boolean, default: !1 },
    iconCustomEmojiId: { type: String }
  },
  emits: ["click"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = W("6.0");
    return i(
      () => t.text,
      (o) => {
        s.setParams({
          text: o
        });
      }
    ), i(
      () => t.color,
      (o) => {
        s.setParams({
          color: o
        });
      }
    ), i(
      () => t.textColor,
      (o) => {
        s.setParams({
          text_color: o
        });
      }
    ), i(
      () => t.visible,
      (o) => {
        s.setParams({
          is_visible: o
        });
      }
    ), i(
      () => t.disabled,
      (o) => {
        s.setParams({
          is_active: !o
        });
      }
    ), i(
      () => t.progress,
      (o) => {
        o ? s.showProgress() : s.hideProgress();
      }
    ), s.isVersionAtLeast("7.10") && i(
      () => t.hasShineEffect,
      (o) => {
        s.hasShineEffect.value = o;
      }
    ), s.isVersionAtLeast("9.5") && i(
      () => t.iconCustomEmojiId,
      (o) => {
        s.iconCustomEmojiId.value = o;
      }
    ), s.onClick(() => a("click")), u(() => {
      C(() => {
        t.progress ? s.showProgress() : s.hideProgress(), s.setParams({
          text: t.text,
          text_color: t.textColor,
          color: t.color,
          has_shine_effect: t.hasShineEffect,
          is_active: !t.disabled,
          is_visible: t.visible,
          icon_custom_emoji_id: t.iconCustomEmojiId
        });
      });
    }), g(() => {
      s.hideProgress(), s.hide();
    }), (o, c) => null;
  }
}), ue = /* @__PURE__ */ r({
  __name: "Popup",
  props: {
    title: { type: String },
    message: { type: String, required: !0 },
    buttons: { type: Array }
  },
  emits: ["close"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = _("6.0");
    return s.isVersionAtLeast("6.2") && u(
      () => s.showPopup(
        {
          title: t.title,
          message: t.message,
          buttons: t.buttons
        },
        (o) => a("close", o)
      )
    ), (o, c) => null;
  }
}), le = /* @__PURE__ */ r({
  __name: "ScanQr",
  props: {
    text: { type: String }
  },
  emits: ["result"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = E("6.0");
    return s.isVersionAtLeast("6.4") && (u(() => {
      s.show(
        {
          text: t.text
        },
        (o) => a("result", o)
      );
    }), g(() => {
      s.close();
    })), (o, c) => null;
  }
}), de = /* @__PURE__ */ r({
  __name: "SecondaryButton",
  props: {
    text: { type: String },
    color: { type: String },
    textColor: { type: String },
    visible: { type: Boolean, default: !0 },
    disabled: { type: Boolean, default: !1 },
    progress: { type: Boolean, default: !1 },
    hasShineEffect: { type: Boolean, default: !1 },
    position: { type: String },
    iconCustomEmojiId: { type: String }
  },
  emits: ["click"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = q("6.0");
    return s.isVersionAtLeast("7.10") && (i(
      () => t.text,
      (o) => {
        s.setParams({
          text: o
        });
      }
    ), i(
      () => t.color,
      (o) => {
        s.setParams({
          color: o
        });
      }
    ), i(
      () => t.textColor,
      (o) => {
        s.setParams({
          text_color: o
        });
      }
    ), i(
      () => t.visible,
      (o) => {
        s.setParams({
          is_visible: o
        });
      }
    ), i(
      () => t.disabled,
      (o) => {
        s.setParams({
          is_active: !o
        });
      }
    ), i(
      () => t.progress,
      (o) => {
        o ? s.showProgress() : s.hideProgress();
      }
    ), i(
      () => t.hasShineEffect,
      (o) => {
        s.hasShineEffect.value = o;
      }
    ), i(
      () => t.position,
      (o) => {
        s.position.value = o;
      }
    ), s.isVersionAtLeast("9.5") && i(
      () => t.iconCustomEmojiId,
      (o) => {
        s.iconCustomEmojiId.value = o;
      }
    ), s.onClick(() => a("click")), u(() => {
      C(() => {
        t.progress ? s.showProgress() : s.hideProgress(), s.setParams({
          text: t.text,
          text_color: t.textColor,
          color: t.color,
          has_shine_effect: t.hasShineEffect,
          position: t.position,
          is_active: !t.disabled,
          is_visible: t.visible,
          icon_custom_emoji_id: t.iconCustomEmojiId
        });
      });
    }), g(() => {
      s.hideProgress(), s.hide();
    })), (o, c) => null;
  }
}), pe = /* @__PURE__ */ r({
  __name: "SettingsButton",
  props: {
    visible: { type: Boolean, default: !0 }
  },
  emits: ["click"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = M("6.0");
    return s.isVersionAtLeast("7.0") && (i(
      () => t.visible,
      (o) => {
        o ? s.show() : s.hide();
      }
    ), s.onClick(() => a("click")), u(() => {
      t.visible && C(() => {
        s.show();
      });
    }), g(() => {
      s.hide();
    })), (o, c) => null;
  }
}), me = /* @__PURE__ */ r({
  __name: "DiscussionWidget",
  props: {
    url: {
      type: String,
      required: !0
    },
    commentsLimit: {
      type: Number,
      default: null
    },
    height: {
      type: String,
      default: null
    },
    color: {
      type: String,
      default: null
    },
    darkColor: {
      type: String,
      default: null
    },
    dark: {
      type: Boolean,
      default: null
    },
    colorful: {
      type: Boolean,
      default: null
    },
    tag: {
      type: String,
      default: "div"
    }
  },
  setup(e) {
    const n = e, t = v(() => JSON.stringify(n));
    return (a, s) => (p(), d(m(e.tag), { key: t.value }, {
      default: A(() => [
        (p(), d(m("script"), {
          async: "",
          src: "https://telegram.org/js/telegram-widget.js?22",
          "data-telegram-discussion": e.url,
          "data-comments-limit": e.commentsLimit,
          "data-height": e.height,
          "data-color": e.color,
          "data-dark-color": e.darkColor,
          "data-dark": e.dark,
          "data-colorful": e.colorful
        }, null, 8, ["data-telegram-discussion", "data-comments-limit", "data-height", "data-color", "data-dark-color", "data-dark", "data-colorful"]))
      ]),
      _: 1
    }));
  }
}), ge = /* @__PURE__ */ r({
  __name: "LoginWidget",
  props: {
    botUsername: {
      type: String,
      required: !0
    },
    redirectUrl: {
      type: String,
      default: null
    },
    size: {
      type: String,
      default: null
    },
    cornerRadius: {
      type: String,
      default: null
    },
    userPhoto: {
      type: Boolean,
      default: null
    },
    requestWrite: {
      type: Boolean,
      default: !1
    },
    tag: {
      type: String,
      default: "div"
    }
  },
  emits: ["auth"],
  setup(e, { emit: n }) {
    const t = e, a = n, s = v(() => JSON.stringify(t));
    return u(() => {
      window.onTelegramAuth = (o) => a("auth", o);
    }), (o, c) => (p(), d(m(e.tag), { key: s.value }, {
      default: A(() => [
        (p(), d(m("script"), {
          async: "",
          src: "https://telegram.org/js/telegram-widget.js?22",
          "data-telegram-login": e.botUsername,
          "data-size": e.size,
          "data-radius": e.cornerRadius,
          "data-userpic": e.userPhoto,
          "data-request-access": e.requestWrite ? "write" : null,
          "data-onauth": e.redirectUrl ? null : "onTelegramAuth(user)",
          "data-auth-url": e.redirectUrl
        }, null, 8, ["data-telegram-login", "data-size", "data-radius", "data-userpic", "data-request-access", "data-onauth", "data-auth-url"]))
      ]),
      _: 1
    }));
  }
}), fe = /* @__PURE__ */ r({
  __name: "PostWidget",
  props: {
    url: {
      type: String,
      required: !0
    },
    width: {
      type: String,
      default: null
    },
    authorPhoto: {
      type: Boolean,
      default: null
    },
    color: {
      type: String,
      default: null
    },
    darkColor: {
      type: String,
      default: null
    },
    dark: {
      type: Boolean,
      default: null
    },
    tag: {
      type: String,
      default: "div"
    }
  },
  setup(e) {
    const n = e, t = v(() => JSON.stringify(n));
    return (a, s) => (p(), d(m(e.tag), { key: t.value }, {
      default: A(() => [
        (p(), d(m("script"), {
          async: "",
          src: "https://telegram.org/js/telegram-widget.js?22",
          "data-telegram-post": e.url,
          "data-width": e.width,
          "data-userpic": e.authorPhoto,
          "data-color": e.color,
          "data-dark-color": e.darkColor,
          "data-dark": e.dark
        }, null, 8, ["data-telegram-post", "data-width", "data-userpic", "data-color", "data-dark-color", "data-dark"]))
      ]),
      _: 1
    }));
  }
}), he = /* @__PURE__ */ r({
  __name: "ShareWidget",
  props: {
    url: {
      type: String,
      required: !0
    },
    comment: {
      type: String,
      default: null
    },
    size: {
      type: String,
      default: null
    },
    noLabel: {
      type: Boolean,
      default: !1
    },
    tag: {
      type: String,
      default: "div"
    }
  },
  setup(e) {
    const n = e, t = v(() => JSON.stringify(n));
    return (a, s) => (p(), d(m(e.tag), { key: t.value }, {
      default: A(() => [
        (p(), d(m("script"), {
          async: "",
          src: "https://telegram.org/js/telegram-widget.js?22",
          "data-telegram-share-url": e.url,
          "data-size": e.size,
          "data-comment": e.comment,
          "data-text": e.noLabel ? "notext" : null
        }, null, 8, ["data-telegram-share-url", "data-size", "data-comment", "data-text"]))
      ]),
      _: 1
    }));
  }
}), Se = {
  ClosingConfirmation: "6.2",
  CloudStorage: "6.9",
  RequestWriteAccess: "6.9",
  RequestContact: "6.9",
  SettingsButton: "7.0",
  BiometricManager: "7.2",
  DisableVerticalSwipes: "7.7"
};
function Be(e) {
  return L().isVersionAtLeast(Se[e]);
}
function Ce() {
  const e = L(), {
    initData: n,
    initDataUnsafe: t,
    version: a,
    isReady: s,
    platform: o,
    isPlatform: c,
    sendData: S,
    ready: B,
    close: y
  } = h("8.0"), l = c("unknown"), b = !l && n === "";
  return {
    initData: n,
    initDataUnsafe: t,
    version: a,
    platform: o,
    isVersionAtLeast: e.isVersionAtLeast,
    sendData: S,
    ready: B,
    close: y,
    isReady: s,
    isPlatform: c,
    isPlatformUnknown: l,
    isFeatureSupported: Be,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onEvent: I,
    /**
     * @deprecated
     */
    canSendData: b
  };
}
function ve() {
  const e = w("8.0");
  return {
    isBackButtonVisible: e.isVisible,
    showBackButton: e.show,
    hideBackButton: e.hide,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onBackButtonClicked: D
  };
}
function Ae() {
  const {
    isInited: e,
    isBiometricAvailable: n,
    biometricType: t,
    isAccessRequested: a,
    isAccessGranted: s,
    deviceId: o,
    isBiometricTokenSaved: c,
    init: S,
    authenticate: B,
    requestAccess: y,
    openSettings: l,
    updateToken: b
  } = P("8.0");
  return {
    isBiometricInited: e,
    isBiometricAvailable: n,
    biometricType: t,
    isBiometricAccessRequested: a,
    isBiometricAccessGranted: s,
    isBiometricTokenSaved: c,
    biometricDeviceId: o,
    initBiometric: S,
    requestBiometricAccess: y,
    authenticateBiometric: B,
    updateBiometricToken: b,
    openBiometricSettings: l,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onBiometricManagerUpdated: $,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onBiometricAuthRequested: F,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onBiometricTokenUpdated: j
  };
}
function _e() {
  const { readText: e } = R("8.0");
  return {
    readTextFromClipboard: e,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onClipboardTextReceived: O
  };
}
function ke() {
  const { isClosingConfirmationEnabled: e } = h("8.0");
  return {
    isClosingConfirmationEnabled: e,
    enableClosingConfirmation() {
      e.value = !0;
    },
    disableClosingConfirmation() {
      e.value = !1;
    }
  };
}
function xe() {
  const e = U("8.0");
  return {
    setStorageItem: e.setItem,
    getStorageItem: e.getItem,
    getStorageItems: e.getItems,
    removeStorageItem: e.removeItem,
    removeStorageItems: e.removeItems,
    getStorageKeys: e.getKeys
  };
}
function we() {
  const { impactOccurred: e, notificationOccurred: n, selectionChanged: t } = Q("8.0");
  return {
    impactOccurred: e,
    notificationOccurred: n,
    selectionChanged: t
  };
}
function Pe() {
  const e = W("8.0");
  return {
    mainButtonText: e.text,
    mainButtonColor: e.color,
    mainButtonTextColor: e.textColor,
    isMainButtonVisible: e.isVisible,
    isMainButtonActive: e.isActive,
    isMainButtonProgressVisible: e.isProgressVisible,
    setMainButtonText(n) {
      e.text.value = n;
    },
    showMainButton: e.show,
    hideMainButton: e.hide,
    enableMainButton: e.enable,
    disableMainButton: e.disable,
    showMainButtonProgress: e.showProgress,
    hideMainButtonProgress: e.hideProgress,
    setMainButtonParams: e.setParams,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onMainButtonClicked: z
  };
}
function Te() {
  const { switchInlineQuery: e, openLink: n, openTelegramLink: t, openInvoice: a } = h("8.0");
  return {
    switchInlineQuery: e,
    openLink: n,
    openTelegramLink: t,
    openInvoice: a,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onInvoiceClosed: H
  };
}
function We() {
  const { showPopup: e, showAlert: n, showConfirm: t } = _("8.0");
  return {
    showPopup: e,
    showAlert: n,
    showConfirm: t,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onPopupClosed: N
  };
}
function Ee() {
  const e = E("8.0");
  return {
    showScanQrPopup: e.show,
    closeScanQrPopup: e.close,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onQrTextReceived: J,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onScanQrPopupClosed: G
  };
}
function Me() {
  const { requestContact: e, requestWriteAccess: n } = h("8.0");
  return {
    requestContact: e,
    requestWriteAccess: n,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onContactRequested: X,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onWriteAccessRequested: K
  };
}
function Le(e, n) {
  const t = n.serialize ?? JSON.stringify, a = x(""), s = (l) => {
    a.value = l, console.error(l);
  }, o = x(!1), { initData: c, initDataUnsafe: S, sendData: B, close: y } = h("8.0");
  return {
    error: a,
    isLoading: o,
    execute: () => {
      B(t(e)), setTimeout(
        () => s(
          "Telegram.WebApp.sendData is only available for custom keyboards."
        ),
        1e3
      );
    },
    executeHttp: async (l, b = {}) => {
      const V = b.closeAfter ?? !0;
      o.value = !0;
      try {
        const f = await fetch(l, {
          method: "POST",
          body: JSON.stringify({
            initData: c,
            initDataUnsafe: S,
            data: t(e)
          })
        });
        return f.ok || s(`${f.status} ${f.statusText}`), f.ok && V && y(), f;
      } catch (f) {
        s(String(f));
      } finally {
        o.value = !1;
      }
    }
  };
}
function Ve() {
  const e = M("8.0");
  return {
    isSettingsButtonVisible: e.isVisible,
    showSettingsButton: e.show,
    hideSettingsButton: e.hide,
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onSettingsButtonClicked: Y
  };
}
function qe() {
  const { shareToStory: e } = h("8.0");
  return {
    shareToStory: e
  };
}
function Ie() {
  const {
    colorScheme: e,
    themeParams: n,
    headerColor: t,
    backgroundColor: a
  } = Z("8.0");
  return {
    colorScheme: e,
    themeParams: n,
    headerColor: t,
    backgroundColor: a,
    setHeaderColor(s) {
      s && (t.value = s);
    },
    setBackgroundColor(s) {
      s && (a.value = s);
    },
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onThemeChanged: ee
  };
}
function De() {
  const {
    isExpanded: e,
    expand: n,
    viewportHeight: t,
    viewportStableHeight: a,
    isVerticalSwipesEnabled: s
  } = k("8.0");
  return {
    isExpanded: e,
    viewportHeight: t,
    viewportStableHeight: a,
    expand: n,
    isVerticalSwipesEnabled: s,
    enableVerticalSwipes() {
      s.value = !0;
    },
    disableVerticalSwipes() {
      s.value = !1;
    },
    /**
     * @deprecated import directly from `vue-tg` instead.
     */
    onViewportChanged: T
  };
}
const je = {
  install(e) {
    e.component("TgAlert", te), e.component("TgBackButton", se), e.component("TgBiometricManager", oe), e.component("TgClosingConfirmation", ne), e.component("TgConfirm", ae), e.component("TgExpandedViewport", ie), e.component("TgFullscreenViewport", re), e.component("TgMainButton", ce), e.component("TgPopup", ue), e.component("TgScanQr", le), e.component("TgSecondaryButton", de), e.component("TgSettingsButton", pe), e.component("TgShareWidget", he), e.component("TgPostWidget", fe), e.component("TgLoginWidget", ge), e.component("TgDiscussionWidget", me);
  }
};
export {
  te as Alert,
  se as BackButton,
  oe as BiometricManager,
  ne as ClosingConfirmation,
  ae as Confirm,
  me as DiscussionWidget,
  ie as ExpandedViewport,
  re as FullscreenViewport,
  ge as LoginWidget,
  ce as MainButton,
  ue as Popup,
  fe as PostWidget,
  le as ScanQr,
  de as SecondaryButton,
  pe as SettingsButton,
  he as ShareWidget,
  je as VueTelegramPlugin,
  Re as isVersionAtLeast,
  Oe as onAccelerometerChanged,
  Ue as onAccelerometerFailed,
  Qe as onAccelerometerStarted,
  ze as onAccelerometerStopped,
  He as onActivated,
  D as onBackButtonClicked,
  F as onBiometricAuthRequested,
  $ as onBiometricManagerUpdated,
  j as onBiometricTokenUpdated,
  O as onClipboardTextReceived,
  X as onContactRequested,
  Ne as onContentSafeAreaChanged,
  Ge as onDeactivated,
  Je as onDeviceOrientationChanged,
  Ke as onDeviceOrientationFailed,
  Xe as onDeviceOrientationStarted,
  Ye as onDeviceOrientationStopped,
  Ze as onEmojiStatusAccessRequested,
  et as onEmojiStatusFailed,
  tt as onEmojiStatusSet,
  I as onEvent,
  st as onFileDownloadRequested,
  ot as onFullscreenChanged,
  nt as onFullscreenFailed,
  at as onGyroscopeChanged,
  it as onGyroscopeFailed,
  rt as onGyroscopeStarted,
  ct as onGyroscopeStopped,
  ut as onHomeScreenAdded,
  lt as onHomeScreenChecked,
  H as onInvoiceClosed,
  dt as onLocationManagerUpdated,
  pt as onLocationRequested,
  z as onMainButtonClicked,
  N as onPopupClosed,
  J as onQrTextReceived,
  mt as onSafeAreaChanged,
  G as onScanQrPopupClosed,
  gt as onSecondaryButtonClicked,
  Y as onSettingsButtonClicked,
  ft as onShareMessageFailed,
  ht as onShareMessageSent,
  ee as onThemeChanged,
  T as onViewportChanged,
  K as onWriteAccessRequested,
  Bt as useAccelerometer,
  yt as useBackButton,
  bt as useBiometricManager,
  Ct as useClipboard,
  vt as useCloudStorage,
  At as useDeviceOrientation,
  _t as useDeviceStorage,
  kt as useEmojiStatus,
  xt as useGyroscope,
  wt as useHapticFeedback,
  Pt as useHomeScreen,
  Tt as useLocationManager,
  Wt as useMainButton,
  Et as useMiniApp,
  Mt as usePopup,
  Lt as useQrScanner,
  Vt as useSecondaryButton,
  qt as useSecureStorage,
  It as useSettingsButton,
  Dt as useTheme,
  jt as useViewport,
  Ce as useWebApp,
  ve as useWebAppBackButton,
  Ae as useWebAppBiometricManager,
  _e as useWebAppClipboard,
  ke as useWebAppClosingConfirmation,
  xe as useWebAppCloudStorage,
  we as useWebAppHapticFeedback,
  Pe as useWebAppMainButton,
  Te as useWebAppNavigation,
  We as useWebAppPopup,
  Ee as useWebAppQrScanner,
  Me as useWebAppRequests,
  Le as useWebAppSendData,
  Ve as useWebAppSettingsButton,
  qe as useWebAppShare,
  Ie as useWebAppTheme,
  De as useWebAppViewport
};
