// vue-tg/dist/9.5.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-CSgDjq9B.mjs";
const y = () => e("9.5"), A = () => s("9.5"), C = () => t("9.5"), D = () => o("9.5"), H = () => u("9.5"), h = () => a("9.5"), j = () => n("9.5"), f = () => c("9.5"), w = () => r("9.5"), E = () => i("9.5"), F = () => $("9.5"), G = () => p("9.5"), L = () => S("9.5"), O = () => g("9.5"), P = () => m("9.5"), Q = () => B("9.5"), T = () => d("9.5"), V = () => M("9.5"), q = () => l("9.5"), x = () => b("9.5"), z = () => k("9.5");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
