// vue-tg/dist/useSecureStorage-CSgDjq9B.mjs
import { getCurrentInstance as ue, onMounted as we, onUnmounted as Ce, effectScope as Be, ref as a, computed as S, readonly as u } from "vue";
function b() {
  if (!("Telegram" in window))
    throw new Error("Telegram SDK not found. Make sure https://telegram.org/js/telegram-web-app.js script is installed.");
  return window?.Telegram?.WebApp;
}
function ke(e) {
  ue() && we(e);
}
function Me(e) {
  ue() && Ce(e);
}
const l = (e, t, n = { manual: !1 }) => {
  const { manual: o } = n, s = b(), r = () => s.onEvent(e, t), i = () => s.offEvent(e, t);
  return o ? r() : (ke(r), Me(i)), {
    off: i
  };
};
function j(e, t) {
  return l("activated", e, t);
}
function V(e, t) {
  return l("deactivated", e, t);
}
function G(e, t) {
  return l("themeChanged", e, t);
}
function H(e, t) {
  return l("viewportChanged", e, t);
}
function $(e, t) {
  return l("safeAreaChanged", e, t);
}
function _(e, t) {
  return l("contentSafeAreaChanged", e, t);
}
function ye(e, t) {
  return l("mainButtonClicked", e, t);
}
function Ie(e, t) {
  return l("secondaryButtonClicked", e, t);
}
function Re(e, t) {
  return l("backButtonClicked", e, t);
}
function Ee(e, t) {
  return l("settingsButtonClicked", e, t);
}
function Ve(e, t) {
  return l("invoiceClosed", e, t);
}
function xe(e, t) {
  return l("popupClosed", e, t);
}
function Pe(e, t) {
  return l("qrTextReceived", e, t);
}
function Fe(e, t) {
  return l("scanQrPopupClosed", e, t);
}
function Oe(e, t) {
  return l("clipboardTextReceived", e, t);
}
function qe(e, t) {
  return l("writeAccessRequested", e, t);
}
function De(e, t) {
  return l("contactRequested", e, t);
}
function U(e, t) {
  return l("biometricManagerUpdated", e, t);
}
function Q(e, t) {
  return l("biometricAuthRequested", e, t);
}
function z(e, t) {
  return l("biometricTokenUpdated", e, t);
}
function W(e, t) {
  return l("fullscreenChanged", e, t);
}
function K(e, t) {
  return l("fullscreenFailed", e, t);
}
function Le(e, t) {
  return l("homeScreenAdded", e, t);
}
function Te(e, t) {
  return l("homeScreenChecked", e, t);
}
function N(e, t) {
  return l("accelerometerStarted", e, t);
}
function J(e, t) {
  return l("accelerometerStopped", e, t);
}
function X(e, t) {
  return l("accelerometerChanged", e, t);
}
function Y(e, t) {
  return l("accelerometerFailed", e, t);
}
function Z(e, t) {
  return l("deviceOrientationStarted", e, t);
}
function ee(e, t) {
  return l("deviceOrientationStopped", e, t);
}
function te(e, t) {
  return l("deviceOrientationChanged", e, t);
}
function ne(e, t) {
  return l("deviceOrientationFailed", e, t);
}
function oe(e, t) {
  return l("gyroscopeStarted", e, t);
}
function se(e, t) {
  return l("gyroscopeStopped", e, t);
}
function re(e, t) {
  return l("gyroscopeChanged", e, t);
}
function ie(e, t) {
  return l("gyroscopeFailed", e, t);
}
function ae(e, t) {
  return l("locationManagerUpdated", e, t);
}
function ce(e, t) {
  return l("locationRequested", e, t);
}
function je(e, t) {
  return l("shareMessageSent", e, t);
}
function Ge(e, t) {
  return l("shareMessageFailed", e, t);
}
function He(e, t) {
  return l("emojiStatusSet", e, t);
}
function $e(e, t) {
  return l("emojiStatusFailed", e, t);
}
function _e(e, t) {
  return l("emojiStatusAccessRequested", e, t);
}
function Ue(e, t) {
  return l("fileDownloadRequested", e, t);
}
function m(e, t) {
  return (...n) => {
    const { hooks: o, omitReturn: s } = t || {};
    o?.before && o.before(n);
    const r = e(...n);
    if (o?.after && o.after(n, r), !s)
      return r;
  };
}
function g(e) {
  return (...t) => new Promise((n) => e(...t, n));
}
function x(e) {
  return (...t) => new Promise((n) => e(...t, n));
}
function C(e) {
  return (...t) => new Promise(
    (n, o) => e(...t, (s, r) => s ? o(s) : n(r))
  );
}
function Qe(e, t) {
  return (...n) => new Promise(
    (o) => e(...n, (...s) => o(t(...s)))
  );
}
function d(e, t) {
  const n = t(e);
  return (...s) => {
    const r = s[s.length - 1];
    if (typeof r == "function") {
      const i = s.slice(0, -1), c = r;
      e(...i, c);
      return;
    }
    return n(...s);
  };
}
function h(e) {
  let t = !1, n;
  return function() {
    return t || (Be(!0).run(() => {
      n = e();
    }), t = !0), n;
  };
}
function le(e, t) {
  const n = (e || "").trim().split(".").map(Number), o = (t || "").trim().split(".").map(Number), s = Math.max(n.length, o.length);
  for (let r = 0; r < s; r++) {
    const i = n[r] || 0, c = o[r] || 0;
    if (i !== c)
      return i > c ? 1 : -1;
  }
  return 0;
}
function pe(e, t) {
  return le(e, t) >= 0;
}
function ze(e, t) {
  return le(e, t) >= 0 ? e : t;
}
function A(e, t, n, o) {
  const s = ze(e, t.version), r = {
    ...n(),
    version: t.version,
    isVersionAtLeast: t.isVersionAtLeast
  };
  for (const i of o)
    pe(s, i.since) && Object.assign(r, i.use());
  return r;
}
const de = h(() => ({
  webApp: b()
}));
function We() {
  const {
    webApp: e
  } = de(), t = d(
    (s, r) => e.showConfirm(s, r),
    g
  ), n = d(
    (s, r) => e.showAlert(s, r),
    x
  ), o = d(
    (s, r) => e.showPopup(s, r),
    g
  );
  return {
    showConfirm: t,
    showAlert: n,
    showPopup: o,
    onClose: xe
  };
}
function jt(e) {
  const { webApp: t } = de();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "6.2", use: We }
    ]
  );
}
const P = h(() => {
  const e = b(), t = a(e.BackButton.isVisible);
  return {
    webApp: e,
    isVisible: t,
    updateState: () => {
      t.value = e.BackButton.isVisible;
    }
  };
});
function Ke() {
  const {
    isVisible: e
  } = P();
  return {
    isVisible: u(e)
  };
}
function Ne() {
  const {
    webApp: e,
    isVisible: t,
    updateState: n
  } = P(), o = m(e.BackButton.show, {
    hooks: {
      after: n
    },
    omitReturn: !0
  }), s = m(e.BackButton.hide, {
    hooks: {
      after: n
    },
    omitReturn: !0
  });
  return {
    isVisible: S({
      get() {
        return t.value;
      },
      set(r) {
        r ? o() : s();
      }
    }),
    show: o,
    hide: s,
    onClick: Re
  };
}
function Gt(e) {
  const { webApp: t } = P();
  return A(
    e,
    t,
    Ke,
    [
      { since: "6.1", use: Ne }
    ]
  );
}
const F = h(() => {
  const e = b(), t = a(e.BiometricManager.isInited), n = a(e.BiometricManager.isBiometricAvailable), o = a(e.BiometricManager.biometricType), s = a(e.BiometricManager.isAccessRequested), r = a(e.BiometricManager.isAccessGranted), i = a(e.BiometricManager.isBiometricTokenSaved), c = a(e.BiometricManager.deviceId);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.BiometricManager.isInited, n.value = e.BiometricManager.isBiometricAvailable, o.value = e.BiometricManager.biometricType, s.value = e.BiometricManager.isAccessRequested, r.value = e.BiometricManager.isAccessGranted, i.value = e.BiometricManager.isBiometricTokenSaved, c.value = e.BiometricManager.deviceId;
    },
    isInited: t,
    isBiometricAvailable: n,
    biometricType: o,
    isAccessRequested: s,
    isAccessGranted: r,
    isBiometricTokenSaved: i,
    deviceId: c
  };
});
function Je() {
  const {
    isInited: e,
    isBiometricAvailable: t,
    biometricType: n,
    isAccessRequested: o,
    isAccessGranted: s,
    isBiometricTokenSaved: r,
    deviceId: i
  } = F();
  return {
    isInited: u(e),
    isBiometricAvailable: u(t),
    biometricType: u(n),
    isAccessRequested: u(o),
    isAccessGranted: u(s),
    isBiometricTokenSaved: u(r),
    deviceId: u(i)
  };
}
function Xe() {
  const {
    webApp: e,
    updateState: t
  } = F(), n = d(
    (c) => e.BiometricManager.init(c),
    x
  ), o = d(
    (c, p) => e.BiometricManager.requestAccess(c, p),
    g
  ), r = d(
    (c, p) => e.BiometricManager.authenticate(c, p),
    (c) => Qe(
      c,
      (p, v) => ({ isAuthenticated: p, token: v })
    )
  ), i = d(
    (c, p) => e.BiometricManager.updateBiometricToken(c, p),
    g
  );
  return U(t), Q(t), z(t), {
    init: n,
    requestAccess: o,
    authenticate: r,
    updateToken: i,
    openSettings: m(e.BiometricManager.openSettings, {
      omitReturn: !0
    }),
    onManagerUpdate: U,
    onAuthRequest: Q,
    onTokenUpdate: z
  };
}
function Ht(e) {
  const { webApp: t } = F();
  return A(
    e,
    t,
    Je,
    [
      { since: "7.2", use: Xe }
    ]
  );
}
const B = h(() => {
  const e = b(), t = a(!1), n = a(e.isClosingConfirmationEnabled), o = a(e.isActive);
  return {
    webApp: e,
    updateState: () => {
      o.value = e.isActive, n.value = e.isClosingConfirmationEnabled;
    },
    isReady: t,
    isActive: o,
    isClosingConfirmationEnabled: n
  };
});
function Ye() {
  const {
    webApp: e,
    isReady: t,
    isActive: n,
    isClosingConfirmationEnabled: o
  } = B();
  return {
    initData: e.initData,
    initDataUnsafe: e.initDataUnsafe,
    platform: e.platform,
    sendData: e.sendData,
    openLink: e.openLink,
    openTelegramLink: e.openTelegramLink,
    ready: m(e.ready, {
      hooks: {
        after: () => {
          t.value = !0;
        }
      }
    }),
    close: e.close,
    isClosingConfirmationEnabled: u(o),
    isActive: u(n),
    isReady: u(t),
    isPlatform: it
  };
}
function Ze() {
  const { webApp: e } = B();
  return {
    openInvoice: d(
      (n, o) => e.openInvoice(n, o),
      g
    ),
    onInvoiceClose: Ve
  };
}
function et() {
  const {
    webApp: e,
    updateState: t,
    isClosingConfirmationEnabled: n
  } = B(), o = m(e.enableClosingConfirmation, {
    hooks: {
      after: t
    }
  }), s = m(e.disableClosingConfirmation, {
    hooks: {
      after: t
    }
  });
  return {
    isClosingConfirmationEnabled: S({
      get() {
        return n.value;
      },
      set(r) {
        r ? o() : s();
      }
    })
  };
}
function tt() {
  const { webApp: e } = B();
  return {
    switchInlineQuery: e.switchInlineQuery
  };
}
function nt() {
  const { webApp: e } = B(), t = d(
    (o) => e.requestContact(o),
    g
  ), n = d(
    (o) => e.requestWriteAccess(o),
    g
  );
  return {
    requestContact: t,
    requestWriteAccess: n,
    onContactRequest: De,
    onWriteAccessRequest: qe
  };
}
function ot() {
  const { webApp: e } = B();
  return {
    shareToStory: e.shareToStory
  };
}
function st() {
  const {
    webApp: e,
    updateState: t
  } = B(), n = d(
    (s, r) => e.shareMessage(s, r),
    g
  ), o = d(
    (s, r) => e.downloadFile(s, r),
    g
  );
  return j(t), V(t), {
    shareMessage: n,
    downloadFile: o,
    onActive: j,
    onDeactivate: V,
    onDeactive: V,
    onShareMessageSent: je,
    onShareMessageFail: Ge,
    onFileDownloadRequest: Ue
  };
}
function rt() {
  const { webApp: e } = B();
  return {
    hideKeyboard: e.hideKeyboard
  };
}
function $t(e) {
  const { webApp: t } = B();
  return A(
    e,
    t,
    Ye,
    [
      { since: "6.1", use: Ze },
      { since: "6.2", use: et },
      { since: "6.7", use: tt },
      { since: "6.9", use: nt },
      { since: "7.8", use: ot },
      { since: "8.0", use: st },
      { since: "9.1", use: rt }
    ]
  );
}
function _t(e) {
  return pe(b().version, e);
}
function it(e) {
  return b().platform === e;
}
const y = h(() => {
  const e = b(), t = a(e.isExpanded), n = a(e.viewportHeight), o = a(e.viewportStableHeight), s = a(e.isFullscreen), r = a(e.isOrientationLocked), i = a(e.isVerticalSwipesEnabled), c = a(e.safeAreaInset), p = a(e.contentSafeAreaInset);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.isExpanded, n.value = e.viewportHeight, o.value = e.viewportStableHeight, s.value = e.isFullscreen, r.value = e.isOrientationLocked, i.value = e.isVerticalSwipesEnabled, c.value = {
        ...e.safeAreaInset
      }, p.value = {
        ...e.contentSafeAreaInset
      };
    },
    isExpanded: t,
    viewportHeight: n,
    viewportStableHeight: o,
    isFullscreen: s,
    isOrientationLocked: r,
    isVerticalSwipesEnabled: i,
    safeAreaInset: c,
    contentSafeAreaInset: p
  };
});
function at() {
  const {
    webApp: e,
    updateState: t,
    isExpanded: n,
    viewportHeight: o,
    viewportStableHeight: s,
    isFullscreen: r,
    isOrientationLocked: i,
    isVerticalSwipesEnabled: c,
    safeAreaInset: p,
    contentSafeAreaInset: v
  } = y();
  return H(t), {
    expand: m(e.expand, {
      hooks: {
        after: t
      }
    }),
    isExpanded: u(n),
    viewportHeight: u(o),
    viewportStableHeight: u(s),
    isFullscreen: u(r),
    isOrientationLocked: u(i),
    isVerticalSwipesEnabled: u(c),
    safeAreaInset: u(p),
    contentSafeAreaInset: u(v),
    onChange: H
  };
}
function ct() {
  const {
    webApp: e,
    isVerticalSwipesEnabled: t,
    updateState: n
  } = y(), o = m(e.enableVerticalSwipes, {
    hooks: {
      after: n
    }
  }), s = m(e.disableVerticalSwipes, {
    hooks: {
      after: n
    }
  });
  return {
    isVerticalSwipesEnabled: S({
      get() {
        return t.value;
      },
      set(r) {
        r ? o() : s();
      }
    })
  };
}
function ut() {
  const {
    webApp: e,
    isFullscreen: t,
    isOrientationLocked: n,
    updateState: o
  } = y(), s = m(e.requestFullscreen, {
    hooks: {
      after: o
    }
  }), r = m(e.exitFullscreen, {
    hooks: {
      after: o
    }
  }), i = m(e.lockOrientation, {
    hooks: {
      after: o
    }
  }), c = m(e.unlockOrientation, {
    hooks: {
      after: o
    }
  });
  return $(o), _(o), W(o), K(o), {
    isFullscreen: S({
      get() {
        return t.value;
      },
      set(p) {
        p ? s() : r();
      }
    }),
    isOrientationLocked: S({
      get() {
        return n.value;
      },
      set(p) {
        p ? i() : c();
      }
    }),
    onSafeAreaChange: $,
    onContentSafeAreaChange: _,
    onFullscreenChange: W,
    onFullscreenFail: K
  };
}
function Ut(e) {
  const { webApp: t } = y();
  return A(
    e,
    t,
    at,
    [
      { since: "7.7", use: ct },
      { since: "8.0", use: ut }
    ]
  );
}
const I = h(() => {
  const e = b(), t = a(e.MainButton.text), n = a(e.MainButton.color), o = a(e.MainButton.textColor), s = a(e.MainButton.isVisible), r = a(e.MainButton.isActive), i = a(e.MainButton.isProgressVisible), c = a(e.MainButton.hasShineEffect), p = a(e.MainButton.iconCustomEmojiId);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.MainButton.text, n.value = e.MainButton.color, o.value = e.MainButton.textColor, s.value = e.MainButton.isVisible, r.value = e.MainButton.isActive, i.value = e.MainButton.isProgressVisible, c.value = e.MainButton.hasShineEffect, p.value = e.MainButton.iconCustomEmojiId;
    },
    text: t,
    color: n,
    textColor: o,
    isVisible: s,
    isActive: r,
    isProgressVisible: i,
    hasShineEffect: c,
    iconCustomEmojiId: p
  };
});
function lt() {
  const {
    webApp: e,
    updateState: t,
    text: n,
    color: o,
    textColor: s,
    isVisible: r,
    isActive: i,
    isProgressVisible: c,
    hasShineEffect: p,
    iconCustomEmojiId: v
  } = I(), k = m(e.MainButton.showProgress, {
    hooks: {
      after: t
    },
    omitReturn: !0
  }), M = m(e.MainButton.hideProgress, {
    hooks: {
      after: t
    },
    omitReturn: !0
  }), w = m(e.MainButton.setParams, {
    hooks: {
      after: t
    },
    omitReturn: !0
  });
  return {
    text: S({
      get() {
        return n.value;
      },
      set(f) {
        w({ text: f });
      }
    }),
    iconCustomEmojiId: u(v),
    color: S({
      get() {
        return o.value;
      },
      set(f) {
        w({ color: f });
      }
    }),
    textColor: S({
      get() {
        return s.value;
      },
      set(f) {
        w({ text_color: f });
      }
    }),
    isVisible: S({
      get() {
        return r.value;
      },
      set(f) {
        w({ is_visible: f });
      }
    }),
    isActive: S({
      get() {
        return i.value;
      },
      set(f) {
        w({ is_active: f });
      }
    }),
    isProgressVisible: S({
      get() {
        return c.value;
      },
      set(f) {
        f ? k() : M();
      }
    }),
    hasShineEffect: u(p),
    show: m(e.MainButton.show, {
      hooks: {
        after: t
      },
      omitReturn: !0
    }),
    hide: m(e.MainButton.hide, {
      hooks: {
        after: t
      },
      omitReturn: !0
    }),
    enable: m(e.MainButton.enable, {
      hooks: {
        after: t
      },
      omitReturn: !0
    }),
    disable: m(e.MainButton.disable, {
      hooks: {
        after: t
      },
      omitReturn: !0
    }),
    showProgress: k,
    hideProgress: M,
    setParams: w,
    onClick: ye
  };
}
function pt() {
  const {
    webApp: e,
    updateState: t,
    hasShineEffect: n
  } = I(), o = m(e.MainButton.setParams, {
    hooks: {
      after: t
    },
    omitReturn: !0
  });
  return {
    hasShineEffect: S({
      get() {
        return n.value;
      },
      set(s) {
        o({ has_shine_effect: s });
      }
    })
  };
}
function dt() {
  const {
    webApp: e,
    updateState: t,
    iconCustomEmojiId: n
  } = I(), o = m(e.MainButton.setParams, {
    hooks: {
      after: t
    },
    omitReturn: !0
  });
  return {
    iconCustomEmojiId: S({
      get() {
        return n.value;
      },
      set(s) {
        o({ icon_custom_emoji_id: s });
      }
    })
  };
}
function Qt(e) {
  const { webApp: t } = I();
  return A(
    e,
    t,
    lt,
    [
      { since: "7.10", use: pt },
      { since: "9.5", use: dt }
    ]
  );
}
const me = h(() => ({
  webApp: b()
}));
function mt() {
  const {
    webApp: e
  } = me();
  return {
    show: d(
      (n, o) => e.showScanQrPopup(n, o),
      g
    ),
    close: e.closeScanQrPopup,
    onScan: Pe
  };
}
function ft() {
  return {
    onClose: Fe
  };
}
function zt(e) {
  const { webApp: t } = me();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "6.4", use: mt },
      { since: "7.7", use: ft }
    ]
  );
}
const R = h(() => {
  const e = b(), t = a(e.SecondaryButton.text), n = a(e.SecondaryButton.color), o = a(e.SecondaryButton.textColor), s = a(e.SecondaryButton.isVisible), r = a(e.SecondaryButton.isActive), i = a(e.SecondaryButton.isProgressVisible), c = a(e.SecondaryButton.hasShineEffect), p = a(e.SecondaryButton.position), v = a(e.SecondaryButton.iconCustomEmojiId);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.SecondaryButton.text, n.value = e.SecondaryButton.color, o.value = e.SecondaryButton.textColor, s.value = e.SecondaryButton.isVisible, r.value = e.SecondaryButton.isActive, i.value = e.SecondaryButton.isProgressVisible, c.value = e.SecondaryButton.hasShineEffect, p.value = e.SecondaryButton.position, v.value = e.SecondaryButton.iconCustomEmojiId;
    },
    text: t,
    color: n,
    textColor: o,
    isVisible: s,
    isActive: r,
    isProgressVisible: i,
    hasShineEffect: c,
    position: p,
    iconCustomEmojiId: v
  };
});
function St() {
  const {
    text: e,
    color: t,
    textColor: n,
    isVisible: o,
    isActive: s,
    isProgressVisible: r,
    hasShineEffect: i,
    position: c,
    iconCustomEmojiId: p
  } = R();
  return {
    text: u(e),
    color: u(t),
    textColor: u(n),
    isVisible: u(o),
    isActive: u(s),
    isProgressVisible: u(r),
    hasShineEffect: u(i),
    position: u(c),
    iconCustomEmojiId: u(p)
  };
}
function bt() {
  const {
    webApp: e,
    text: t,
    color: n,
    textColor: o,
    isVisible: s,
    isActive: r,
    isProgressVisible: i,
    hasShineEffect: c,
    position: p,
    updateState: v
  } = R(), k = m(e.SecondaryButton.showProgress, {
    hooks: {
      after: v
    },
    omitReturn: !0
  }), M = m(e.SecondaryButton.hideProgress, {
    hooks: {
      after: v
    },
    omitReturn: !0
  }), w = m(e.SecondaryButton.setParams, {
    hooks: {
      after: v
    },
    omitReturn: !0
  });
  return {
    text: S({
      get() {
        return t.value;
      },
      set(f) {
        w({ text: f });
      }
    }),
    color: S({
      get() {
        return n.value;
      },
      set(f) {
        w({ color: f });
      }
    }),
    textColor: S({
      get() {
        return o.value;
      },
      set(f) {
        w({ text_color: f });
      }
    }),
    isVisible: S({
      get() {
        return s.value;
      },
      set(f) {
        w({ is_visible: f });
      }
    }),
    isActive: S({
      get() {
        return r.value;
      },
      set(f) {
        w({ is_active: f });
      }
    }),
    isProgressVisible: S({
      get() {
        return i.value;
      },
      set(f) {
        f ? k() : M();
      }
    }),
    hasShineEffect: S({
      get() {
        return c.value;
      },
      set(f) {
        w({ has_shine_effect: f });
      }
    }),
    position: S({
      get() {
        return p.value;
      },
      set(f) {
        w({ position: f });
      }
    }),
    show: m(e.SecondaryButton.show, {
      hooks: {
        after: v
      },
      omitReturn: !0
    }),
    hide: m(e.SecondaryButton.hide, {
      hooks: {
        after: v
      },
      omitReturn: !0
    }),
    enable: m(e.SecondaryButton.enable, {
      hooks: {
        after: v
      },
      omitReturn: !0
    }),
    disable: m(e.SecondaryButton.disable, {
      hooks: {
        after: v
      },
      omitReturn: !0
    }),
    showProgress: k,
    hideProgress: M,
    setParams: w,
    onClick: Ie
  };
}
function gt() {
  const {
    webApp: e,
    updateState: t,
    iconCustomEmojiId: n
  } = R(), o = m(e.SecondaryButton.setParams, {
    hooks: {
      after: t
    },
    omitReturn: !0
  });
  return {
    iconCustomEmojiId: S({
      get() {
        return n.value;
      },
      set(s) {
        o({ icon_custom_emoji_id: s });
      }
    })
  };
}
function Wt(e) {
  const { webApp: t } = R();
  return A(
    e,
    t,
    St,
    [
      { since: "7.10", use: bt },
      { since: "9.5", use: gt }
    ]
  );
}
const O = h(() => {
  const e = b(), t = a(e.SettingsButton.isVisible);
  return {
    webApp: e,
    isVisible: t,
    updateState: () => {
      t.value = e.SettingsButton.isVisible;
    }
  };
});
function ht() {
  const {
    isVisible: e
  } = O();
  return {
    isVisible: u(e)
  };
}
function At() {
  const {
    webApp: e,
    isVisible: t,
    updateState: n
  } = O(), o = m(e.SettingsButton.show, {
    hooks: {
      after: n
    },
    omitReturn: !0
  }), s = m(e.SettingsButton.hide, {
    hooks: {
      after: n
    },
    omitReturn: !0
  });
  return {
    isVisible: S({
      get() {
        return t.value;
      },
      set(r) {
        r ? o() : s();
      }
    }),
    show: o,
    hide: s,
    onClick: Ee
  };
}
function Kt(e) {
  const { webApp: t } = O();
  return A(
    e,
    t,
    ht,
    [
      { since: "7.0", use: At }
    ]
  );
}
const fe = h(() => ({
  webApp: b()
}));
function vt() {
  const {
    webApp: e
  } = fe();
  return {
    readText: d(
      (n) => e.readTextFromClipboard(n),
      g
    ),
    onReadText: Oe
  };
}
function Nt(e) {
  const { webApp: t } = fe();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "6.4", use: vt }
    ]
  );
}
const Se = h(() => ({
  webApp: b()
}));
function wt() {
  const {
    webApp: e
  } = Se(), t = d(
    (c, p, v) => e.CloudStorage.setItem(c, p, v),
    C
  ), n = d(
    (c, p) => e.CloudStorage.getItem(c, p),
    C
  ), o = d(
    (c, p) => e.CloudStorage.getItems(c, p),
    C
  ), s = d(
    (c, p) => e.CloudStorage.removeItem(c, p),
    C
  ), r = d(
    (c, p) => e.CloudStorage.removeItems(c, p),
    C
  ), i = d(
    (c) => e.CloudStorage.getKeys(c),
    C
  );
  return {
    setItem: t,
    getItem: n,
    getItems: o,
    removeItem: s,
    removeItems: r,
    getKeys: i
  };
}
function Jt(e) {
  const { webApp: t } = Se();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "6.9", use: wt }
    ]
  );
}
const be = h(() => ({
  webApp: b()
}));
function Ct() {
  const {
    webApp: e
  } = be(), t = m(e.HapticFeedback.impactOccurred, {
    omitReturn: !0
  }), n = m(e.HapticFeedback.notificationOccurred, {
    omitReturn: !0
  }), o = m(e.HapticFeedback.selectionChanged, {
    omitReturn: !0
  });
  return {
    impactOccurred: t,
    notificationOccurred: n,
    selectionChanged: o
  };
}
function Xt(e) {
  const { webApp: t } = be();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "6.1", use: Ct }
    ]
  );
}
const E = h(() => {
  const e = b(), t = a(e.colorScheme), n = a(e.themeParams), o = a(e.headerColor), s = a(e.backgroundColor), r = a(e.bottomBarColor);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.colorScheme, n.value = {
        ...e.themeParams
      }, o.value = e.headerColor, s.value = e.backgroundColor, r.value = e.bottomBarColor;
    },
    colorScheme: t,
    themeParams: n,
    headerColor: o,
    backgroundColor: s,
    bottomBarColor: r
  };
});
function Bt() {
  const {
    colorScheme: e,
    themeParams: t,
    headerColor: n,
    backgroundColor: o,
    bottomBarColor: s,
    updateState: r
  } = E();
  return G(r), {
    colorScheme: u(e),
    themeParams: u(t),
    headerColor: u(n),
    backgroundColor: u(o),
    bottomBarColor: u(s),
    onChange: G
  };
}
function kt() {
  const {
    webApp: e,
    headerColor: t,
    backgroundColor: n,
    updateState: o
  } = E(), s = m(e.setHeaderColor, {
    hooks: {
      after: o
    }
  }), r = m(e.setBackgroundColor, {
    hooks: {
      after: o
    }
  });
  return {
    headerColor: S({
      get() {
        return t.value;
      },
      set(i) {
        s(i);
      }
    }),
    backgroundColor: S({
      get() {
        return n.value;
      },
      set(i) {
        r(i);
      }
    })
  };
}
function Mt() {
  const {
    webApp: e,
    bottomBarColor: t,
    updateState: n
  } = E(), o = m(e.setBottomBarColor, {
    hooks: {
      after: n
    }
  });
  return {
    bottomBarColor: S({
      get() {
        return t.value;
      },
      set(s) {
        o(s);
      }
    })
  };
}
function Yt(e) {
  const { webApp: t } = E();
  return A(
    e,
    t,
    Bt,
    [
      { since: "6.1", use: kt },
      { since: "7.10", use: Mt }
    ]
  );
}
const q = h(() => {
  const e = b(), t = a(e.Accelerometer.isStarted), n = a(e.Accelerometer.x), o = a(e.Accelerometer.y), s = a(e.Accelerometer.z);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.Accelerometer.isStarted, n.value = e.Accelerometer.x, o.value = e.Accelerometer.y, s.value = e.Accelerometer.z;
    },
    isStarted: t,
    x: n,
    y: o,
    z: s
  };
});
function yt() {
  const {
    isStarted: e,
    x: t,
    y: n,
    z: o
  } = q();
  return {
    isStarted: u(e),
    x: u(t),
    y: u(n),
    z: u(o)
  };
}
function It() {
  const {
    webApp: e,
    updateState: t
  } = q(), n = d(
    (s, r) => e.Accelerometer.start(s, r),
    g
  ), o = d(
    (s) => e.Accelerometer.stop(s),
    g
  );
  return N(t), J(t), X(t), Y(t), {
    start: n,
    stop: o,
    onStart: N,
    onStop: J,
    onChange: X,
    onFail: Y
  };
}
function Zt(e) {
  const { webApp: t } = q();
  return A(
    e,
    t,
    yt,
    [
      { since: "8.0", use: It }
    ]
  );
}
const D = h(() => {
  const e = b(), t = a(e.DeviceOrientation.isStarted), n = a(e.DeviceOrientation.absolute), o = a(e.DeviceOrientation.alpha), s = a(e.DeviceOrientation.beta), r = a(e.DeviceOrientation.gamma);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.DeviceOrientation.isStarted, n.value = e.DeviceOrientation.absolute, o.value = e.DeviceOrientation.alpha, s.value = e.DeviceOrientation.beta, r.value = e.DeviceOrientation.gamma;
    },
    isStarted: t,
    absolute: n,
    alpha: o,
    beta: s,
    gamma: r
  };
});
function Rt() {
  const {
    isStarted: e,
    absolute: t,
    alpha: n,
    beta: o,
    gamma: s
  } = D();
  return {
    isStarted: u(e),
    absolute: u(t),
    alpha: u(n),
    beta: u(o),
    gamma: u(s)
  };
}
function Et() {
  const {
    webApp: e,
    updateState: t
  } = D(), n = d(
    (s, r) => e.DeviceOrientation.start(s, r),
    g
  ), o = d(
    (s) => e.DeviceOrientation.stop(s),
    g
  );
  return Z(t), ee(t), te(t), ne(t), {
    start: n,
    stop: o,
    onStart: Z,
    onStop: ee,
    onChange: te,
    onFail: ne
  };
}
function en(e) {
  const { webApp: t } = D();
  return A(
    e,
    t,
    Rt,
    [
      { since: "8.0", use: Et }
    ]
  );
}
const ge = h(() => ({
  webApp: b()
}));
function Vt() {
  const {
    webApp: e
  } = ge(), t = d(
    (r, i, c) => e.DeviceStorage.setItem(r, i, c),
    C
  ), n = d(
    (r, i) => e.DeviceStorage.getItem(r, i),
    C
  ), o = d(
    (r, i) => e.DeviceStorage.removeItem(r, i),
    C
  ), s = d(
    (r) => e.DeviceStorage.clear(r),
    g
  );
  return {
    setItem: t,
    getItem: n,
    removeItem: o,
    clear: s
  };
}
function tn(e) {
  const { webApp: t } = ge();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "9.0", use: Vt }
    ]
  );
}
const he = h(() => ({
  webApp: b()
}));
function xt() {
  const {
    webApp: e
  } = he(), t = d(
    (o, s, r) => e.setEmojiStatus(o, s, r),
    g
  ), n = d(
    (o) => e.requestEmojiStatusAccess(o),
    g
  );
  return {
    set: t,
    requestAccess: n,
    onSet: He,
    onFail: $e,
    onAccessRequest: _e
  };
}
function nn(e) {
  const { webApp: t } = he();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "8.0", use: xt }
    ]
  );
}
const L = h(() => {
  const e = b(), t = a(e.Gyroscope.isStarted), n = a(e.Gyroscope.x), o = a(e.Gyroscope.y), s = a(e.Gyroscope.z);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.Gyroscope.isStarted, n.value = e.Gyroscope.x, o.value = e.Gyroscope.y, s.value = e.Gyroscope.z;
    },
    isStarted: t,
    x: n,
    y: o,
    z: s
  };
});
function Pt() {
  const {
    isStarted: e,
    x: t,
    y: n,
    z: o
  } = L();
  return {
    isStarted: u(e),
    x: u(t),
    y: u(n),
    z: u(o)
  };
}
function Ft() {
  const {
    webApp: e,
    updateState: t
  } = L(), n = d(
    (s, r) => e.Gyroscope.start(s, r),
    g
  ), o = d(
    (s) => e.Gyroscope.stop(s),
    g
  );
  return oe(t), se(t), re(t), ie(t), {
    start: n,
    stop: o,
    onStart: oe,
    onStop: se,
    onChange: re,
    onFail: ie
  };
}
function on(e) {
  const { webApp: t } = L();
  return A(
    e,
    t,
    Pt,
    [
      { since: "8.0", use: Ft }
    ]
  );
}
const Ae = h(() => ({
  webApp: b()
}));
function Ot() {
  const {
    webApp: e
  } = Ae(), t = d(
    (n) => e.checkHomeScreenStatus(n),
    g
  );
  return {
    addShortcut: e.addToHomeScreen,
    checkShortcutStatus: t,
    onShortcutAdd: Le,
    onShortcutCheck: Te
  };
}
function sn(e) {
  const { webApp: t } = Ae();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "8.0", use: Ot }
    ]
  );
}
const T = h(() => {
  const e = b(), t = a(e.LocationManager.isInited), n = a(e.LocationManager.isLocationAvailable), o = a(e.LocationManager.isAccessRequested), s = a(e.LocationManager.isAccessGranted);
  return {
    webApp: e,
    updateState: () => {
      t.value = e.LocationManager.isInited, n.value = e.LocationManager.isLocationAvailable, o.value = e.LocationManager.isAccessRequested, s.value = e.LocationManager.isAccessGranted;
    },
    isInited: t,
    isLocationAvailable: n,
    isAccessRequested: o,
    isAccessGranted: s
  };
});
function qt() {
  const {
    isInited: e,
    isLocationAvailable: t,
    isAccessRequested: n,
    isAccessGranted: o
  } = T();
  return {
    isInited: u(e),
    isLocationAvailable: u(t),
    isAccessRequested: u(n),
    isAccessGranted: u(o)
  };
}
function Dt() {
  const {
    webApp: e,
    updateState: t
  } = T(), n = d(
    (s) => e.LocationManager.init(s),
    x
  ), o = d(
    (s) => e.LocationManager.getLocation(s),
    g
  );
  return ae(t), ce(t), {
    init: n,
    getLocation: o,
    openSettings: m(e.LocationManager.openSettings, {
      omitReturn: !0
    }),
    onManagerUpdate: ae,
    onRequest: ce
  };
}
function rn(e) {
  const { webApp: t } = T();
  return A(
    e,
    t,
    qt,
    [
      { since: "8.0", use: Dt }
    ]
  );
}
const ve = h(() => ({
  webApp: b()
}));
function Lt() {
  const {
    webApp: e
  } = ve(), t = d(
    (i, c, p) => e.SecureStorage.setItem(i, c, p),
    C
  ), n = d(
    (i, c) => e.SecureStorage.getItem(i, c),
    C
  ), o = d(
    (i, c) => e.SecureStorage.restoreItem(i, c),
    C
  ), s = d(
    (i, c) => e.SecureStorage.removeItem(i, c),
    C
  ), r = d(
    (i) => e.SecureStorage.clear(i),
    g
  );
  return {
    setItem: t,
    getItem: n,
    restoreItem: o,
    removeItem: s,
    clear: r
  };
}
function an(e) {
  const { webApp: t } = ve();
  return A(
    e,
    t,
    () => ({}),
    [
      { since: "9.0", use: Lt }
    ]
  );
}
export {
  ee as $,
  Q as A,
  U as B,
  Oe as C,
  ye as D,
  Ve as E,
  xe as F,
  Fe as G,
  Pe as H,
  qe as I,
  De as J,
  Ee as K,
  G as L,
  _t as M,
  j as N,
  V as O,
  $ as P,
  _ as Q,
  Ie as R,
  W as S,
  K as T,
  Le as U,
  Te as V,
  N as W,
  J as X,
  X as Y,
  Y as Z,
  Z as _,
  Gt as a,
  te as a0,
  ne as a1,
  oe as a2,
  se as a3,
  re as a4,
  ie as a5,
  ae as a6,
  ce as a7,
  je as a8,
  Ge as a9,
  He as aa,
  $e as ab,
  _e as ac,
  Ue as ad,
  Ht as b,
  Nt as c,
  Jt as d,
  en as e,
  tn as f,
  nn as g,
  on as h,
  Xt as i,
  sn as j,
  rn as k,
  Qt as l,
  $t as m,
  jt as n,
  zt as o,
  Wt as p,
  an as q,
  Kt as r,
  Yt as s,
  Ut as t,
  Zt as u,
  H as v,
  b as w,
  l as x,
  Re as y,
  z
};
